nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/AbstractJMSProcessor.java

```diff
@@ -118,6 +118,23 @@
             .defaultValue(Charset.defaultCharset().name())
             .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
             .build();
+    static final PropertyDescriptor ALLOW_ILLEGAL_HEADER_CHARS = new PropertyDescriptor.Builder()
+            .name("Allow Illegal Characters in Header Names")
+            .description("Specifies whether illegal characters in header names should be sent to the JMS broker. " +
+                    "Usually hyphens and full-stops.")
+            .required(true)
+            .defaultValue("false")
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor ATTRIBUTES_AS_HEADERS_REGEX = new PropertyDescriptor.Builder()
+            .name("Attributes to Send as JMS Headers (Regex)")
+            .description("Specifies the Regular Expression that determines the names of FlowFile attributes that" +
+                    " should be sent as JMS Headers")
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .defaultValue(".*")
+            .required(false)
```

Comment:
I think it's better to make it required. Or check for null in the code otherwise.